src/components/PartySwitch/PartySwitch.tsx

```diff
@@ -77,6 +78,10 @@ export const PartySwitch = ({
   const [selectedId, setSelectedId] = useState(currentPartyId);
   const [open, setOpen] = useState(false);
   const [filter, setFilter] = useState("");
+  const [visibleParties, setVisibleParties] = useState<Array<PartySwitchItem>>(
+    []
+  );
+  const [containerRef, setContainerRef] = useState<HTMLDivElement | null>(null);
```

Comment:
Why not `useRef`? 
Your approach works, but it generates an unnecessary rerender while setting the container ref.

```suggestion
const containerRef = useRef<HTMLDivElement>(null);
```

Reply:
The decision to use useState instead of useRef was made due to encountering an issue with useRef. This would cause the ref to be null when u open the Drawer component and have a value when you close the Drawer. Using useState instead of useRef was deemed a better choice in this scenario due to the specific requirements of the component. In our case, the container reference may not be immediately available, as it is asynchronously set after the component mounts.